lib/TPP/Dialect/Xsmm/XsmmOps.cpp

```diff
@@ -16,3 +16,82 @@
 
 using namespace mlir;
 using namespace mlir::xsmm;
+
+template <typename EnumClass>
+static ParseResult parseEnum(EnumClass &value, OpAsmParser &parser) {
+  StringRef flag;
+  auto loc = parser.getCurrentLocation();
+  if (parser.parseKeyword(&flag))
+    return failure();
+  auto flagAttr = symbolizeEnum<EnumClass>(flag);
+  if (!flagAttr)
+    return parser.emitError(loc, "invalid enum") << flag;
+  value = *flagAttr;
+  return success();
+}
+
+static ParseResult parserImpl(OpAsmParser &parser, OperationState &result) {
+  auto &builder = parser.getBuilder();
+  // Parse the input
+  result.addAttribute("inputs", DenseI64ArrayAttr::parse(parser, Type{}));
```

Comment:
nit: this hits assert on any unexpected token - `parseCustomAttributeWithFallback` (or sth else) would be more graceful

Reply:
You mean `DenseI64ArrayAttr::parse`?

Reply:
Yes